crates/node/src/network_server/metrics.rs

```diff
@@ -188,38 +188,43 @@ pub async fn render_metrics(State(state): State<NodeCtrlHandlerState>) -> String
     let manager = RocksDbManager::get();
     let all_dbs = manager.get_all_dbs();
 
-    let mut global_labels: Vec<String> = vec![];
-    for (k, v) in state.prometheus_handle.global_labels() {
-        global_labels.push(format!("{}=\"{}\"", k, formatting::sanitize_label_value(v)));
-    }
+    let global_labels = state.prometheus_handle.global_labels();
+    let global_labels_len = global_labels.len();
+    let mut labels = Vec::with_capacity(global_labels_len + 2);
+    labels.clone_from(global_labels);
+    // Push an empty db、cf label at the end of labels,
+    // so in the for loop below, we can modify for each db label.
+    labels.extend(vec![String::from("db=\"\""), String::from("cf=\"\"")]);
 
     // Overall write buffer manager stats
     format_rocksdb_property_for_prometheus(
         &mut out,
-        &global_labels,
+        &labels[..global_labels_len],
         MetricUnit::Bytes,
         "rocksdb.memory.write_buffer_manager_capacity",
         manager.get_total_write_buffer_capacity(),
     );
 
     format_rocksdb_property_for_prometheus(
         &mut out,
-        &global_labels,
+        &labels[..global_labels_len],
         MetricUnit::Bytes,
         "rocksdb.memory.write_buffer_manager_usage",
         manager.get_total_write_buffer_usage(),
     );
 
     for db in &all_dbs {
-        let mut labels = Vec::with_capacity(global_labels.len() + 1);
-        global_labels.clone_into(&mut labels);
-        labels.push(format!(
-            "db=\"{}\"",
-            formatting::sanitize_label_value(&db.name)
-        ));
+        labels[global_labels_len] =
+            format!("db=\"{}\"", formatting::sanitize_label_value(&db.name));
```

Comment:
If we add `labels.push(format!("db=\"{}\"", formatting::sanitize_label_value(&db.name))` here and at the end of the loop `labels.pop()`, then we don't have to extend the array before with empty strings.